reference/hooks.rst

```diff
@@ -0,0 +1,100 @@
+.. _hooks_reference:
```

Comment:
Remove all this file, and move important bits to the ``extensions/hooks`` file.

Reply:
Done, it has been removed.